akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -88,22 +88,27 @@ private[http] object Http2Blueprint {
       settings: ServerSettings,
       log: LoggingAdapter,
       initialDemuxerSettings: immutable.Seq[Setting] = Nil,
-      upgraded: Boolean = false): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] =
-    httpLayer(settings, log) atop
+      upgraded: Boolean = false,
+      telemetry: TelemetrySpi): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
+    telemetry.serverConnection atop
```

Comment:
This is what made me think of akka/akka#29901 for when its not enabled

Reply:
That optimization would be good in any case for other parts of implementation stacks.